src/main/java/com/gmail/kutilandrej/cinema/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.gmail.kutilandrej.cinema.dao.impl;
+
+import com.gmail.kutilandrej.cinema.dao.ShoppingCartDao;
+import com.gmail.kutilandrej.cinema.exception.DataProcessingException;
+import com.gmail.kutilandrej.cinema.lib.Dao;
+import com.gmail.kutilandrej.cinema.model.ShoppingCart;
+import com.gmail.kutilandrej.cinema.model.User;
+import com.gmail.kutilandrej.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert ShoppingCart entity", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            String hql = "FROM ShoppingCart SC LEFT JOIN FETCH SC.user "
```

Comment:
do we need to fetch user here?

Reply:
don't necessary, we have user and we don't need to fetch him. I'll fix it.